src/action_cc.cc

```diff
@@ -295,43 +295,79 @@ NM_ACTION_(nickel_setting) {
     #define NM_ERR_RET nullptr
 
-    if (!strcmp(arg, "web_browser")) {
-        //libnickel 4.6 * _ZN26N3SettingsExtrasControllerC2Ev
-        void (*N3SettingsExtrasController_N3SettingsExtrasController)(N3SettingsExtrasController*);
-        reinterpret_cast<void*&>(N3SettingsExtrasController_N3SettingsExtrasController) = dlsym(RTLD_DEFAULT, "_ZN26N3SettingsExtrasControllerC2Ev");
-        NM_ASSERT(N3SettingsExtrasController_N3SettingsExtrasController, "could not dlsym N3SettingsExtrasController constructor");
-
-        //libnickel 4.6 * _ZN26N3SettingsExtrasControllerD1Ev
-        void (*N3SettingsExtrasController_N3SettingsExtrasControllerD)(N3SettingsExtrasController*);
-        reinterpret_cast<void*&>(N3SettingsExtrasController_N3SettingsExtrasControllerD) = dlsym(RTLD_DEFAULT, "_ZN26N3SettingsExtrasControllerD1Ev");
-        NM_ASSERT(N3SettingsExtrasController_N3SettingsExtrasControllerD, "could not dlsym N3SettingsExtrasController destructor");
-
-        //libnickel 4.6 * _ZN26N3SettingsExtrasController11openBrowserEv
-        void (*N3SettingsExtrasController_openBrowser)(N3SettingsExtrasController*);
-        reinterpret_cast<void*&>(N3SettingsExtrasController_openBrowser) = dlsym(RTLD_DEFAULT, "_ZN26N3SettingsExtrasController11openBrowserEv");
-        NM_ASSERT(N3SettingsExtrasController_openBrowser, "could not dlsym BrowserWorkflowManager::openBrowser");
-
-        N3SettingsExtrasController *nse = alloca(128); // as of 4.20.14622, it's actually 48 bytes, but we're going to stay on the safe side
-        N3SettingsExtrasController_N3SettingsExtrasController(nse);
-        N3SettingsExtrasController_openBrowser(nse);
-        N3SettingsExtrasController_N3SettingsExtrasControllerD(nse);
-
-        // the QObject is only used to pass events to it, but there's something I'm missing here which leads to it segfaulting after connecting to WiFi if it isn't already connected
-        /*
-        //libnickel 4.11.11911 * _ZN22BrowserWorkflowManagerC1EP7QObject
-        void (*BrowserWorkflowManager_BrowserWorkflowManager)(BrowserWorkflowManager*, QObject*);
+    if (!strncmp(arg, "web_browser", 11)) {
+        bool modal;
+        QUrl *url;
+        QString *css;
+
+        if (!strcmp(arg, "web_browser")) {
+            modal = false;
+            url = new QUrl();
+            css = new QString("");
+        } else {
+            char *tmp1 = strdupa(arg); // strsep and strtrim will modify it
+            char *arg1 = strtrim(strsep(&tmp1, ":"));
+            char *arg2 = strtrim(tmp1);
+
+            if (!arg2 || strcmp(arg1, "web_browser"))
+                NM_RETURN_ERR("unknown beta feature name or plugin mimetype '%s' (split: '%s')", arg, arg1);
+
+            QString tmp = QString::fromUtf8(arg2).trimmed();
+
+            if (tmp.section(':', 0, 0).trimmed() == "modal") {
+                modal = true;
+                tmp = tmp.section(':', 1).trimmed();
+            } else {
+                modal = false;
+            }
```

Comment:
```suggestion
        bool modal = false;
        QUrl *url;
        QString *css;

        if (!strcmp(arg, "web_browser")) {
            url = new QUrl();
            css = new QString("");
        } else {
            char *tmp1 = strdupa(arg); // strsep and strtrim will modify it
            char *arg1 = strtrim(strsep(&tmp1, ":"));
            char *arg2 = strtrim(tmp1);

            if (!arg2 || strcmp(arg1, "web_browser"))
                NM_RETURN_ERR("unknown beta feature name or plugin mimetype '%s' (split: '%s')", arg, arg1);

            QString tmp = QString::fromUtf8(arg2).trimmed();

            if (tmp.section(':', 0, 0).trimmed() == "modal") {
                modal = true;
                tmp = tmp.section(':', 1).trimmed();
            }
```
May as well make the model setting false by default. Saves a like or three of code.

Sorry about the noise. Still trying to figure this code suggestion stuff out.

Reply:
The main reason why I decided to keep it this way is so the compiler/static analyzer will warn if I missed any possibilities in the if statements.

And, you did do the code suggestion correctly, although it's generally better to submit multiple small ones instead of one large one (they can be batched when being applied).

Reply:
That's alright.

And yeah, makes sense with the static analysis.